sw/device/tests/rstmgr_cpu_info_test.c

```diff
@@ -0,0 +1,202 @@
+// Copyright lowRISC contributors.
+// Licensed under the Apache License, Version 2.0, see LICENSE for details.
+// SPDX-License-Identifier: Apache-2.0
+
+#include "sw/device/lib/base/abs_mmio.h"
+#include "sw/device/lib/base/mmio.h"
+#include "sw/device/lib/dif/dif_pwrmgr.h"
+#include "sw/device/lib/dif/dif_rstmgr.h"
+#include "sw/device/lib/runtime/ibex.h"
+#include "sw/device/lib/runtime/log.h"
+#include "sw/device/lib/testing/aon_timer_testutils.h"
+#include "sw/device/lib/testing/flash_ctrl_testutils.h"
+#include "sw/device/lib/testing/rstmgr_testutils.h"
+#include "sw/device/lib/testing/test_framework/check.h"
+#include "sw/device/lib/testing/test_framework/ottf_isrs.h"
+#include "sw/device/lib/testing/test_framework/ottf_macros.h"
+#include "sw/device/lib/testing/test_framework/ottf_main.h"
+
+#include "hw/top_earlgrey/sw/autogen/top_earlgrey.h"
+
+/**
+ * RSTMGR CPU INFO TEST
+ *  This test create double fault by accessing register with
+ *  non-existing address.
+ *  After double fault, dut got reset by watch dog bite
+ *  and test collect / check cpu_info from rstmgr
+ */
+
+// non existing address
+#define kIllegalAddr1 0x4041FFF0u
+#define kIllegalAddr2 0x40003618u
+#define kSkipComp 0x12345678u
+#define kCpuDumpSize 9
+
+/**
+ * Cpu dump struct index
+ */
+enum {
+  kCpuDumpIdxCurrentExceptionAddr = 0,
+  kCpuDumpIdxCurrentLastDataAddr = 1,
+  kCpuDumpIdxCurrentNextPc = 2,
+  kCpuDumpIdxCurrentPc = 3,
+  kCpuDumpIdxPreviousExceptionAddr = 4,
+  kCpuDumpIdxPreviousLastDataAddr = 5,
+  kCpuDumpIdxPreviousNextPc = 6,
+  kCpuDumpIdxPreviousPc = 7,
+  kCpuDumpIdxPreviousValid = 8,
+};
+
+/**
+ * Reserve expected cpu dump area in flash
+ */
+__attribute__((section(".non_volatile_scratch")))
+const volatile dif_rstmgr_cpu_info_dump_segment_t exp_dump[kCpuDumpSize] = {
+    UINT32_MAX, UINT32_MAX, UINT32_MAX, UINT32_MAX, UINT32_MAX,
+    UINT32_MAX, UINT32_MAX, UINT32_MAX, UINT32_MAX};
+
+/**
+ *  Dump structure:
+ *    0: current.exception_addr
+ *    1: current.last_data_addr
+ *   *2: current.next_pc
+ *   *3: current.pc
+ *    4: previous.exception_addr
+ *    5: previous.last_data_addr
+ *   *6: previous.next_pc
+ *    7: previous.pc
+ *    8: previous_valid
+ *
+ * Observed cpu dump will be collected after watch dog bite,
+ * exp cpu dump will be created at the 'ottf_exception_handler'.
+ * Following fields are current code specific and
+ * will be skipped comparison.
+ *
+ *  current.next_pc, current.pc, previous.next_pc
+ */
+static dif_rstmgr_cpu_info_dump_segment_t dump[kCpuDumpSize];
+static dif_rstmgr_cpu_info_dump_segment_t temp_dump[kCpuDumpSize] = {
+    kSkipComp, kSkipComp, kSkipComp, kSkipComp, kSkipComp,
+    kSkipComp, kSkipComp, kSkipComp, kSkipComp};
+
+static dif_flash_ctrl_state_t flash_ctrl;
+
+// Count number of falut
+static uint32_t global_error_cnt;
```

Comment:
you might need to make this volatile.
This could explain why your illegal address didn't change.

Reply:
rerun with a static volatile uint32_t global_error_cnt;
result doesn't change.